src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs

```diff
@@ -61,23 +59,47 @@ private UpgradeStepInitializeResult InitializeImpl(IUpgradeContext context)
             }
 
             var projects = context.Projects.ToList();
-
-            if (projects.All(p => IsCompleted(context, p)))
+            var completeChecks = projects.Select(async p => IsCompleted(context, p) || !await RunChecksAsync(p, token).ConfigureAwait(false));
+            if ((await Task.WhenAll(completeChecks).ConfigureAwait(false)).All(b => b))
             {
                 return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "No projects need upgrade", BuildBreakRisk.None);
             }
 
+            IProject? newCurrentProject = null;
+
             if (projects.Count == 1)
             {
-                var project = projects[0];
-                context.SetCurrentProject(project);
-
-                Logger.LogInformation("Setting only project in solution as the current project: {Project}", project.FilePath);
+                newCurrentProject = projects[0];
+                Logger.LogDebug("Setting only project in solution as the current project: {Project}", newCurrentProject.FilePath);
+            }
+            else if (!context.InputIsSolution)
+            {
+                // If the user has specified a particular project, only that should be the current project
+                newCurrentProject = projects.Where(i => Path.GetFileName(i.FilePath).Equals(Path.GetFileName(context.InputPath), StringComparison.OrdinalIgnoreCase)).First();
+                Logger.LogDebug("Setting user-selected project as the current project: {Project}", newCurrentProject.FilePath);
+            }
 
-                return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected only project.", BuildBreakRisk.None);
+            if (newCurrentProject is null)
+            {
+                return new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, "No project is currently selected.", BuildBreakRisk.None);
```

Comment:
What's the user experience when `CurrentProjectSelectionStep` is `Incomplete`? Does it look and ask the user to re-run?

Reply:
It will show the user a list of projects and have them pick the one they want to upgrade.